keras_nlp/utils/text_generation.py

```diff
@@ -48,7 +41,20 @@ def validate_token_probability_fn(token_probability_fn, prompt):
         )
 
 
-def mask_tokens_after_end_token(prompt, max_length, end_token_id, pad_token_id):
+def _validate_inputs(token_probability_fn, prompt):
+    prompt = _validate_prompt(prompt)
+
+    input_is_1d = prompt.shape.rank == 1
+    if input_is_1d:
+        prompt = prompt[tf.newaxis, :]
+
+    _validate_token_probability_fn(token_probability_fn, prompt)
+    return prompt, input_is_1d
+
+
+def _mask_tokens_after_end_token(
```

Comment:
@mattdangerw I remember we explicitly wanted these to be public functions, do you remember the reason?